app/models/shipit/stack.rb

```diff
@@ -607,5 +606,19 @@ def emit_merge_status_hooks
     def ci_enabled_cache_key
       "stacks:#{id}:ci_enabled"
     end
+
+    def should_delay_continuous_delivery?(commit)
+      commit.deploy_failed? ||
+        (checks? && !EphemeralCommitChecks.new(commit).run.success?) ||
+        commit.recently_pushed? ||
+        cached_deploy_spec.config.empty?
+    end
```

Comment:
This should be a performance improvement:

```suggestion
    def should_delay_continuous_delivery?(commit)
      cached_deploy_spec.config.empty? ||
        commit.deploy_failed? ||
        (checks? && !EphemeralCommitChecks.new(commit).run.success?) ||
        commit.recently_pushed?
    end
```